uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/logging/Logger.java

```diff
@@ -0,0 +1,160 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.logging;
+
+import androidx.annotation.NonNull;
+
+/**
+ * An abstract logger to write/send logs. The actual implementation of logger used may differ.
+ * The logger must be registered with the {@link LoggerRegistry} to use that logging implementation.
+ */
+public class Logger {
```

Comment:
If this became a LoggerFactory that generated ILogger instances on a per-tag basis that delegated to all registers appenders (potentially for that tag) you'd have something that looked a lot like log4j and java.util.Logging.

Reply:
As discussed on chat the other day....not moving for factory pattern yet....keeping static references for now....we could do this later in a separate PR if needed.